src/sentry/discover/endpoints/discover_key_transactions.py

```diff
@@ -43,51 +45,143 @@ def get(self, request, organization):
+                )
+                if len(keyed_transactions) == len(data["team"]):
                     return Response(status=204)
 
+                keyed_transaction_team_ids = {k.team_id for k in keyed_transactions}
+
                 try:
-                    KeyTransaction.objects.create(**base_filter)
+                    # TeamKeyTransaction.objects.create(**base_filter)
+                    TeamKeyTransaction.objects.bulk_create(
+                        [
+                            TeamKeyTransaction(**base_filter, team=team)
+                            for team in data["team"]
+                            if team.id not in keyed_transaction_team_ids
+                        ]
+                    )
                     return Response(status=201)
-                # Even though we tried to avoid it, this KeyTransaction was created already
+                # Even though we tried to avoid it, the TeamKeyTransaction was created already
                 except IntegrityError:
-                    return Response(status=204)
-            return Response(serializer.errors, status=400)
+                    return Response(status=409)
+
+        return Response(serializer.errors, status=400)
 
     def delete(self, request, organization):
         """ Remove a Key transaction for a user """
         if not self.has_feature(request, organization):
             return Response(status=404)
 
         project = self.get_project(request, organization)
-        transaction = request.data["transaction"]
 
-        try:
-            model = KeyTransaction.objects.get(
-                transaction=transaction,
-                organization=organization,
-                project=project,
-                owner=request.user,
-            )
-        except KeyTransaction.DoesNotExist:
+        if not self.has_team_feature(request, organization):
+            transaction = request.data["transaction"]
+            try:
+                model = KeyTransaction.objects.get(
+                    transaction=transaction,
+                    organization=organization,
+                    project=project,
+                    owner=request.user,
+                )
+            except KeyTransaction.DoesNotExist:
+                return Response(status=204)
+
+            model.delete()
+
+            return Response(status=204)
+
+        serializer = serializers.TeamKeyTransactionSerializer(
+            data=request.data, context={"request": request}
+        )
+
+        if serializer.is_valid():
+            data = serializer.validated_data
+            base_filter = {
+                "organization": organization,
+                "project": project,
+                "transaction": data["transaction"],
+            }
+
+            try:
+                model = TeamKeyTransaction.objects.filter(
+                    **base_filter, team__in=data["team"]
+                ).delete()
+            except TeamKeyTransaction.DoesNotExist:
```

Comment:
I don't think `filter(...).delete()` would ever raise a DoesNotExist, so you can probably skip the try/except

```
In [5]: TeamKeyTransaction.objects.filter(team=team)
Out[5]: <BaseQuerySet []>

In [6]: TeamKeyTransaction.objects.filter(team=team).delete()
Out[6]: (0, {'sentry.TeamKeyTransaction': 0})
```

Reply:
removed in b9be6be